packages/noya-state/src/reducers/application.ts

```diff
@@ -853,6 +871,116 @@ export function reducer(
         }
       });
     }
+    case 'addColorSwatch': {
+      return produce(state, (state) => {
+        const sharedSwatches = state.sketch.document.sharedSwatches;
+
+        const swatchColor: Sketch.Color = {
+          "_class": "color",
+          "alpha": 0.5,
+          "red": 0,
+          "green": 0,
+          "blue": 0
+        }
+        const swatch: Sketch.Swatch = {
+          "_class": "swatch",
+          "do_objectID": uuid(),
+          "name": "Undefined",
+          "value": swatchColor,
+        }
+
+        sharedSwatches?.objects.push(swatch);
+        state.selectedSwatchIds = [swatch.do_objectID];
+      });
+    }
+    case 'selectSwatch': {
+      const [, id, selectionType = 'replace'] = action;
+
+      const ids = id === undefined ? [] : typeof id === 'string' ? [id] : id;
+
+      return produce(state, (state) => {
+        switch (selectionType) {
+          case 'intersection':
+            state.selectedSwatchIds.push(
+              ...ids.filter((id) => !state.selectedSwatchIds.includes(id)),
+            );
+            return;
+          case 'difference':
+            ids.forEach((id) => {
+              const selectedIndex = state.selectedSwatchIds.indexOf(id);
+              state.selectedSwatchIds.splice(selectedIndex, 1);
+            });
+            return;
+          case 'replace':
+            state.selectedSwatchIds = [...ids];
+            return;
+        }
+      });
+    }
+    case 'setSwatchColor': {
+      const [, swatchId, color] = action;
```

Comment:
We should always "normalize" the swatchId to an array here so the rest of the code is simpler.

E.g. `const swatchIds = typeof swatchId === 'string' ? [swatchId] : swatchId`;

Reply:
Normalized